Sources/Datadog/RUMMonitor.swift

```diff
@@ -248,53 +248,87 @@ public class RUMMonitor: DDRUMMonitor, RUMCommandSubscriber {
         )
     }
 
-    override public func startResourceLoading(resourceKey: String, url: URL, httpMethod: RUMHTTPMethod, attributes: [AttributeKey: AttributeValue]) {
+    override public func startResourceLoading(
+        resourceKey: String,
+        request: URLRequest,
```

Comment:
`request` parameter seems confusing to me
we use it only for detecting `.xhr` kind, doesn't its `ResourceScope` already know that?
shouldn't we assume that request is immutable and the one that starts `ResourceScope` will finish it?

Reply:
I think you refer to `stopResourceLoading(...)`:
```swift
override public func stopResourceLoading(
    resourceKey: String,
    response: URLResponse,
    request: URLRequest? = nil,
    size: Int64? = nil,
    attributes: [AttributeKey: AttributeValue]
)
```
Indeed, that parameter, although optional, looks weird 👍. 

I refactored the `RUMResourceScope` a bit, so I managed to remove it. Now when the resource is started, we capture "kind by request" (optional), and when the resource is finished, we capture "kind by response" (non-optional). When sending the actual event, the "by request" takes precedence before the "by response".

With this approach, we will be compatible with upcoming bridge interface for RN, where `kind:` will be explicitly given in public API. I wanted to avoid passing `URLRequest` and `URLResponse` to scopes, as those will not exist in RN.